### PR TITLE
feat(sdk-core): add ecdsa to verifyShareProof()

### PR DESCRIPTION
**Pull request by brekiy**
Adds an ECDSA path to the verifyShareProof function to be used for verifying
ECDSA keyshares.

Ticket: BG-65151

BREAKING CHANGE: The `verifyEdShareProof` function is renamed to
`verifyShareProof` and now expects a 4th parameter which must be a
string of either `eddsa` or `ecdsa`.

### CODE REVIEW

File: modules/sdk-core/src/bitgo/utils/opengpgUtils.ts [hunk 1]
```diff
@@ -61,6 +61,30 @@ export async function verifyEdShareProof(
...
 
+/**
+ * Verify an Ecdsa KeyShare Proof.
+ *
+ * @param senderPubKey public key of the sender of the privateShareProof
+ * @param privateShareProof u value proof
+ * @param uValue u value from an Ecdsa keyshare
+ * @return {boolean} whether uValue proof actually was signed by sender as part of their subkeys
+ */
+export async function verifyEcShareProof(
```
Comment (jicli) [inline modules/sdk-core/src/bitgo/utils/opengpgUtils.ts:72]: since majority of the verify is the same 
we can do with `verifyShareProof(..., algo)`?

Comment (brekiy) [reply modules/sdk-core/src/bitgo/utils/opengpgUtils.ts:72]: good point

File: modules/sdk-core/src/bitgo/utils/opengpgUtils.ts [hunk 2]
```diff
@@ -36,29 +36,39 @@ export async function getBitgoGpgPubKey(bitgo: BitGoBase): Promise<Key> {
...
-  return decodedUValueProof === rawUValueProof;
+  if (algo === 'eddsa') {
+    const decodedUValueProof = Buffer.from(proofSubkeys.keyPacket.publicParams['Q'].slice(1)).toString('hex');
+    const rawUValueProof = Buffer.from(
+      sodium.crypto_scalarmult_ed25519_base_noclamp(Buffer.from(uValue, 'hex'))
+    ).toString('hex');
+    return decodedUValueProof === rawUValueProof;
+  } else if (algo === 'ecdsa') {
+    const decodedUValueProof = Buffer.from(proofSubkeys.keyPacket.publicParams['Q']).toString('hex');
+    const rawUValueProof = secp256k1.pointFromScalar(Buffer.from(uValue, 'hex'), false);
```
Comment (Logicwax) [inline modules/sdk-core/src/bitgo/utils/opengpgUtils.ts:67]: is `secp256k1.publicKeyCreate()` not available?

Comment (brekiy) [reply modules/sdk-core/src/bitgo/utils/opengpgUtils.ts:67]: Not in the used library (`modules/utxo-lib-src/noble_ecc.ts`)

Comment (brekiy) [reply modules/sdk-core/src/bitgo/utils/opengpgUtils.ts:67]: This is `pointFromScalar()`'s definition:
https://github.com/BitGo/BitGoJS/blob/master/modules/utxo-lib/src/noble_ecc.ts#L50-L51

It uses `getPublicKey()` from `noble-secp256k1`.
https://www.npmjs.com/package/@noble/secp256k1#getpublickeyprivatekey

**Review by Logicwax — APPROVED** [review]

**Comment by brekiy** [comment]
Rebased to fix merge conflict

**Review by islamaminBitGo — APPROVED** [review]

**Review by byhow — CHANGES_REQUESTED** [review]
Changes LGTM; breaking changes for extra parameters that are added to the function needs to have `BREAKING:` in the commit message unless this function is not public facing

**Comment by islamaminBitGo** [comment]
> Changes LGTM; breaking changes for extra parameters that are added to the function needs to have `BREAKING:` in the commit message unless this function is not public facing

i can confirm that it is public facing as im able to use it outside of the sdk. I agree with adding `BREAKING:` in the commit message

**Review by byhow — APPROVED** [review]
Appreciate the change!

**Pull request merged**